components/object_store/src/obkv/util.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use table_kv::{KeyBoundary, ScanRequest};
+
+/// Generate ScanRequest with prefix
+pub fn scan_request_with_prefix(prefix_bytes: &[u8]) -> ScanRequest {
+    let mut start_key = Vec::with_capacity(prefix_bytes.len());
+    start_key.extend(prefix_bytes);
+    let start = KeyBoundary::included(start_key.as_ref());
+
+    let mut end_key = Vec::with_capacity(prefix_bytes.len());
+    end_key.extend(prefix_bytes);
+    let carry = add_one(&mut end_key);
+    // Check add one operation overflow.
+    let end = if carry == 1 {
+        KeyBoundary::MaxIncluded
+    } else {
+        KeyBoundary::excluded(end_key.as_ref())
+    };
+    table_kv::ScanRequest {
+        start,
+        end,
+        reverse: false,
+    }
+}
+
+pub fn add_one(nums: &mut [u8]) -> u8 {
```

Comment:
I guess there is no need to make it public:
```suggestion
/// Increment one to the byte array, and return the carry.
fn inc_by_one(nums: &mut [u8]) -> u8 {
```

Reply:
Yes, it can be private.